src/folding/hypernova/circuit.rs

```diff
@@ -1,2 +1,105 @@
 // hypernova nimfs verifier circuit
 // see section 5 in https://eprint.iacr.org/2023/573.pdf
+
+use ark_ff::PrimeField;
+use ark_r1cs_std::{
+    fields::{fp::FpVar, FieldVar},
+    ToBitsGadget,
+};
+use ark_relations::r1cs::SynthesisError;
+use std::marker::PhantomData;
+
+/// Gadget to compute the sum of all $\gamma^{j} \cdot eq(r_{x_j}, r_x^{\prime}) \cdot \sigma_j$.
+pub struct SumMulsGammaPowsEqSigmaGadget<F: PrimeField> {
+    _f: PhantomData<F>,
+}
+
+impl<F: PrimeField> SumMulsGammaPowsEqSigmaGadget<F> {
+    /// Computes the sum $\Sigma_{j}^{j + n} \gamma^{j} \cdot eq_eval \cdot \sigma_{j}$, where $n$ is the length of the `sigmas` vector
+    /// It corresponds to the first term of the sum that $\mathcal{V}$ has to compute at section 5, step 5 of "A multi-folding scheme for CCS".
+    ///
+    /// # Arguments
+    /// - `sigmas`: vector of $\sigma_j$ values
+    /// - `eq_eval`: the value of $\tilde{eq}(x_j, x^{\prime})$
+    /// - `gamma`: a `GammaVar`, which supports a `pow` method, representing $\gamma$
```

Comment:
Maybe now the doc for `gamma` can be updated to remove the `GammaVar` reference

Reply:
Ah good catch! [Done]([06806c8](https://github.com/privacy-scaling-explorations/folding-schemes/pull/53/commits/06806c8d4fb686ee31e4b52bb3eace90a6902647))